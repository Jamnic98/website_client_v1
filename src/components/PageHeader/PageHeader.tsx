import React from 'react'

import styles from './pageHeader.module.css'

interface PageHeaderProps {
  title: string
  description?: string
}

export const PageHeader: React.FC<PageHeaderProps> = ({ title, description }) => (
  <header className={styles.header} data-testid="application-pageHeader">
    <h1 className={styles.title}>{title}</h1>
    <p className={styles.description}>{description}</p>
    <hr className={styles.hr} />
  </header>
)
