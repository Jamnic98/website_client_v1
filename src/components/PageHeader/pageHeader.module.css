.description {
  text-align: left;
}

.hr {
  width: 100%;
  height: 1px;
  border-style: none;
  background-color: var(--c-bg-secondary);
}
