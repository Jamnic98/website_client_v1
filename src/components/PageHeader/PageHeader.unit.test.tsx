import React from 'react'
import { render, screen } from '@testing-library/react'

import { PageHeader } from './PageHeader'

const testTitle = 'Title'
const testDescription = 'This is a test.'

describe('test PageHeader component', () => {
  it('renders correctly', () => {
    render(<PageHeader title={testTitle} description={testDescription} />)
    const header = screen.getByTestId('application-pageHeader')
    expect(header.getElementsByTagName('h1')[0]).toHaveTextContent(testTitle)
    expect(header.getElementsByTagName('p')[0]).toHaveTextContent(
      testDescription
    )
  })
})
