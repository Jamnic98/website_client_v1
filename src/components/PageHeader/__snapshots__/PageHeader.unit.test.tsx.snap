// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PageHeader component renders PageHeader unchanged 1`] = `
<header
  className="header"
  data-testid="application-pageHeader"
>
  <h1
    className="title"
  >
    Title
  </h1>
  <p
    className="description"
  >
    This is a test.
  </p>
  <hr
    className="hr"
  />
</header>
`;
